app/models/ext_management_system.rb

```diff
@@ -904,14 +920,14 @@ def stop_event_monitor_queue
   end
 
   def stop_event_monitor_queue_on_change
-    if event_monitor_class && !self.new_record? && default_endpoint.changed.include_any?("hostname", "ipaddress")
+    if event_monitor_class && !new_record? && default_endpoint.changed.include_any?("hostname", "ipaddress")
       _log.info("EMS: [#{name}], Hostname or IP address has changed, stopping Event Monitor.  It will be restarted by the WorkerMonitor.")
       stop_event_monitor_queue
     end
   end
 
   def stop_event_monitor_queue_on_credential_change
-    if event_monitor_class && !self.new_record? && self.credentials_changed?
```

Comment:
NOTE `credentials_changed?` used an instance variable set early in `#update_authentications` but it seemed easier to just check `default_authentication.changed?`

I didn't want to rewrite the `credentials_changed?` method to use this in case something else depended on that exact behavior.

Also that method only checked `auth_user_pwd` which wouldn't work for k8s/gce/etc... which use an auth_key.

I would like to be able to specify e.g. `authentication_for_events` rather than `default_authentication` in case a provider doesn't use the default_auth for events.

Reply:
Is that something you want to do as a followup or for this PR?

Reply:
I've added `authtype_for_event/refresh` and `endpoint_role_for_event/refresh` so that the callbacks can check the changes for endpoints they care about